pkg/eval/realloader.go

```diff
@@ -166,7 +212,10 @@ func (c *client) discover(discovery discoveryclient.DiscoveryInterface) error {
 			if apiRes.Namespaced {
 				c.nsResources = append(c.nsResources, res)
 			} else {
-				c.nonNsResources = append(c.nonNsResources, res)
+				c.nonNsResources[schema.GroupResource{
+					Group:    gv.Group,
+					Resource: apiRes.Name,
+				}] = gv.Version
```

Comment:
I am wondering if the namespace information is necessary here. Additionally I think we know all the information at this point (during discovery) - group, version, resource and kind so perhaps we could use only one map and remove the `gvrCache`. It's just an idea....I haven't tried yet.

Reply:
We're still using the nsResources to limit which resources we're searching through, depending on the `ns == ""` condition. The semantics of `ns == ""` is to search on nonNs resources and vice-verse.  https://github.com/iNecas/kube-health/blob/main/pkg/eval/realloader.go#L199

Reply:
It might be worth having the GR -> version map regardless of the namespaces, and then keep the ns/non-ns slices separate, or perhaps having an internal struct (`GRmeta` or something) that would carry all this additional info (what's the version, is it namespaced), and then having functions around it to give GVR filtered on the ns/non-ns criteria?